### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by kevinslin**
Various changes to documentation:
- Added design principles in README
- Added feature list to README
- sync root README with plugin-core README
- add new README in `/docs/`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,62 +1,166 @@
...
 
-A personal knowledge management tool that **grows** as you do.
+Dendron is an **open-source, local-first, markdown-based, note-taking tool**. It's a personal knowledge management solution (PKM) built specifically for developers and integrates natively with IDEs like [VSCode](https://code.visualstudio.com/).
 
-Dendron is a local-first, markdown based, hierarchical note-taking application built on top of [VSCode](https://code.visualstudio.com/) and [friends](./ACKNOWLEDGEMENTS.md).
+## Why
 
-Whether you organize your notes with [PARA](https://fortelabs.co/blog/para/) or [Zettelkasten](https://zettelkasten.de/), link notes together like [roam](https://roamresearch.com/), or simply create notes in an ad-hoc manner, Dendron can help.
+Most PKM tools will help you create notes but hit a wall trying to retrieve them past some threshold. That threshold varies with the tool — 100, 1000, 10 000, virtually everything stops working past 10k notes unless the human was super diligent about organizing their knowledge. Past this threshold, entropy wins and every query becomes a keyword search and scrolling through pages of results.
```
Comment (jonathanyeung) [inline README.md:14]: Some grammatical things:

Most PKM tools help you create notes but hit limitations when trying to retrieve notes once your knowledge base reaches a certain size threshold. That threshold varies with the tool, but virtually everything stops working past 10k notes unless the user was extremely diligent about organizing their knowledge. Past this threshold, entropy wins and every query becomes a keyword search and scrolling through pages of results.

File: README.md [hunk 2]
```diff
@@ -1,62 +1,166 @@
...
 
-Think of it as a [second brain](https://www.buildingasecondbrain.com/) to help you **make sense** of all the information you care about.
+> Dendron's mission is to help **humans organize, find, and work with any amount of knowledge**.
 
-Some potential use cases include:
+It not only helps you create notes, but also retrieve them - retrieval works as well with ten notes as it does with ten thousand.
```
Comment (jonathanyeung) [inline README.md:18]: grammar nit: no comma after `notes`

File: README.md [hunk 3]
```diff
@@ -1,62 +1,166 @@
...
 
-- project management
-- personal knowledge base of things you've learned
-- CRM to keep track of clients, friends and enemies
-- recipe book
+## How
 
-Click [here](https://youtu.be/6rwWUalr9Ac) to see an example of Dendron in action.
+Dendron builds on top of the past five decades of language design and developer tooling. We apply the key lessons from software to the management of general knowledge. We make managing general knowledge more like managing code, and your PKM more like an IDE for general knowledge.
```
Comment (jonathanyeung) [inline README.md:22]: `of *programming* language design`

"We make managing general knowledge like managing code and your PKM like an IDE."

File: README.md [hunk 4]
```diff
@@ -1,62 +1,166 @@
...
+## Design Principles
 
-Dendron can be installed as a [vscode plugin](https://marketplace.visualstudio.com/items?itemName=dendron.dendron). If you don't already have VSCode, you can download it [here](https://code.visualstudio.com/).
+### Developer Centric
 
-After you have downloaded the plugin, following the instructions to create your first **workspace**.
+Dendron aims to create a world class developer experience for managing knowledge.
 
-1. Launch the _command bar_:
+Our goal is to provide a tool with the efficiency of vim, the extensibility of emacs, and the approcahbility of VSCode.
```
Comment (jonathanyeung) [inline README.md:30]: sp on approachability

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (kevinslin) [reply README.md:14]: fixed grammar, kept wall analogy

**Pull request closed without merging**